internal/worker/modelworkermanager/package_test.go

```diff
@@ -9,7 +9,7 @@ import (
 	gc "gopkg.in/check.v1"
 )
 
-//go:generate go run go.uber.org/mock/mockgen -typed -package modelworkermanager_test -destination domainservices_mock_test.go github.com/juju/juju/internal/services DomainServices,DomainServicesGetter
+//go:generate go run go.uber.org/mock/mockgen -typed -package modelworkermanager_test -destination domainservices_mock_test.go github.com/juju/juju/internal/services DomainServices,DomainServicesGetter,ControllerDomainServices
```

Comment:
Can we not use these big interfaces to avoid generating those huge mock files?

Reply:
Hi Kelvin, ControllerDomainServices is used in `func (m *modelWorkerManager) loop() error `

Reply:
See my other comment - no need to mock the entire services factory.

Reply:
I see, thanks for the clarification ian!